__tests__/rojo-thinks.spec.js

```diff
@@ -0,0 +1,131 @@
+        describe('when the string also contains "time"', () => {
+          it('should return a string containing am or pm', () => {
+            const timeRegex = /.+am|pm.*/i;
+            const testString = 'What time are you coming home?';
+            const result = timeRegex.test(test.subject(testString));
+
+            expect(result).toBe(true);
+          });
+        });
+
+        describe('when the string also contains "age"', () => {
+          it('should return a string containing an age', () => {
+            const ageRegex = /^\d{1,2}$/;
+            const testString = 'What age are you?';
+            const result = ageRegex.test(test.subject(testString));
+
+            expect(result).toBe(true);
+          });
+        });
+
+        describe('when the string also contains "day"', () => {
+          it('should return a string containing a day of the week', () => {
+            const dayRegex = /.*monday|tuesday|wednesday|thursday|friday|saturday|sunday.*/i;
+            const testString = 'What day are you coming home?';
+            const result = dayRegex.test(test.subject(testString));
+
+            expect(result).toBe(true);
+          });
+        });
+        
+        describe('when the string does not contain an answerable "what" question', () => {
+          it('should return a string of what?', () => {
+            const whatRegex = /.*what*/i;
+            const testString = 'What colour are you?';
+            const result = whatRegex.test(test.subject(testString));
+
+            expect(result).toBe(true);
+          });
+        });
+     }); // what
+
+      // where
+      describe('when passed a string that starts with "where"', () => {
+        describe('when the string also contains "my"', () => {
+          it('should return a string of a location name?', () => {
+            const placeRegex = /.*bed|fridge|sofa|table*/i;
+            const testString = 'Where are my keys?';
+            const result = placeRegex.test(test.subject(testString));
+
+            expect(result).toBe(true);
+          });
+        });
+
+        describe('when the string does not contain an answerable "where" question', () => {
+          it('should return a string of where?', () => {
+            const whereRegex = /.*where*/i;
+            const testString = 'Where is?';
+            const result = whereRegex.test(test.subject(testString));
+
+            expect(result).toBe(true);
+          });
+        });
+      }); //where
+
+      // which
+      describe('when passed a string that starts with "which"', () => {
+        describe('when the string also contains "country"', () => {
+          it('should return a string of a country name?', () => {
+            const countryRegex = /.*Mexico|Brazil|France|Italy|Spain*/i;
+            const testString = 'Which country shall we go to?';
+            const result = countryRegex.test(test.subject(testString));
+
+            expect(result).toBe(true);
```

Comment:
Once you get better with tests, the preferred pattern is to put setup in a `beforeEach`, and then your `it` block will only contain assertions. Example:

```js
beforeEach(() => {
  test.regex = /.*Mexico|Brazil|France|Italy|Spain*/i;
  test.result = test.subject('Which country shall we go to?');
});

it('should return a string of a country name?', () => {
  expect(test.regex.test(test.result)).toBe(true);
});
```